ash-bergs/dish-dollar
# [frontend] feat: pantry drawer

### Add Pantry Drawer with Animations

#### Description:
This PR introduces the Pantry Drawer, an expandable and collapsible section below each pantry card that will contain interactive layers for each pantry item. To enhance the user experience, smooth animations are added for the drawer's appearance and disappearance.

#### Changes:

1. **PantryCard Component Enhancements**: The main pantry card now has a click event to toggle the visibility of the drawer. This will be updated in future iterations, clicking on the PantryActions will trigger this.
   
2. **Introduction of MotionBox**: Leveraged `framer-motion` library to create a `MotionBox` - a motion-enhanced version of Chakra's `Box` component. This component is responsible for the drawer's smooth animations.

3. **Drawer Animations**:
   - **Entrance**: Uses a spring-based transition for a bouncy effect as it appears.
   - **Exit**: Smoothly slides out of view when the card or drawer is clicked again.
  
4. **Use of AnimatePresence**: To manage the mounting and unmounting of the drawer with animations, `AnimatePresence` from `framer-motion` is utilized.

#### Testing:

To test the new feature:
1. Navigate to the pantry list.
2. Click on any pantry card. Observe a drawer sliding down with additional details.
3. Click on the card again. The drawer should smoothly slide out of view.

#### Future Considerations:
- The current animation properties are set with initial values. Fine-tuning might be necessary based on user feedback or design requirements.
- Next up we'll need to add the interactive elements that will actually go here - i.e. the slider and the edit item form 

<img width="895" alt="image" src="https://github.com/ash-bergs/dish-dollar/assets/65979049/939defce-8b81-4867-b014-d7c70cc1dca9">
